web/src/integTest/java/com/microsoft/applicationinsights/framework/ApplicationTelemetryManager.java

```diff
@@ -0,0 +1,154 @@
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.framework;
+
+import com.microsoft.applicationinsights.framework.telemetries.DocumentType;
+import com.microsoft.applicationinsights.framework.telemetries.TelemetryItem;
+import com.microsoft.applicationinsights.framework.telemetries.TelemetryItemFactory;
+import com.microsoft.applicationinsights.internal.util.LocalStringsUtils;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.queue.CloudQueueMessage;
+import org.apache.commons.lang3.time.StopWatch;
+import org.json.JSONObject;
+
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Created by yonisha on 6/16/2015.
+ */
+public class ApplicationTelemetryManager {
+
+    private static final String RUN_ID_QUERY_PARAM_NAME = "runId";
+    private static final int POLLING_TIMEOUT_IN_SECONDS = 180;
+    private static final int QUEUE_POLLING_INTERVAL_IN_SECONDS = 3;
+    private static final int MS_IN_SECOND = 1000;
+
+    private List<TelemetryItem> cachedTelemetries = new ArrayList<TelemetryItem>();
+    private ApplicationTelemetryQueue applicationTelemetryQueue;
+    private ApplicationBlobClient applicationBlobClient;
+
+    public ApplicationTelemetryManager(String storageAccountConnectionString, String queueName) throws StorageException, InvalidKeyException, URISyntaxException {
+        this.applicationTelemetryQueue = new ApplicationTelemetryQueue(storageAccountConnectionString, queueName);
+        this.applicationBlobClient = new ApplicationBlobClient(storageAccountConnectionString);
+    }
+
+    public List<TelemetryItem> getApplicationTelemetries(String runId, int numberOfExpectedTelemetries) throws Exception {
+
+        List<TelemetryItem> telemetriesForRunId = getTelemetriesFromCacheWithRunId(runId);
+
+        if (telemetriesForRunId.size() < numberOfExpectedTelemetries) {
```

Comment:
It would be good also the check the other case `telemetriesForRunId.size() > numberOfExpectedTelemetries` for the case that for some reason a telemetry is being sent twice (we had such bug in the past)